File: frontend/app/src/main/java/com/example/caffwebshop/model/CommentPublic.kt
package com.example.caffwebshop.model

import com.google.gson.annotations.SerializedName

data class CommentPublic (
    @SerializedName("id")
    val id: Int,
    @SerializedName("content")
    val content: String,
    @SerializedName("createdAt")
    val createdAt: String,
    @SerializedName("authorId")
    val authorId: Int,
    @SerializedName("caffItemId")
    val caffItemId: Int,
    @SerializedName("author")
    val author: UserPublic
)

File: backend/CaffShop/Models/Comment/CommentPublic.cs
using System;
using CaffShop.Models.User;

namespace CaffShop.Models.Comment
{
    public class CommentPublic
    {
        public long Id { get; set; }
        public string Content { get; set; }
        public DateTime CreatedAt { get; set; }
        public long AuthorId { get; set; }
        public long CaffItemId { get; set; }
        public UserPublic Author { get; set; }
    }
}